### ISSUE TITLE
MutableStateFieldCheck: "equalsChanged" misleading

### ISSUE THREAD

**Issue by C-Otto** [jqno/equalsverifier#315]
Up front: I'm dealing with legacy code, which is why my bug report might be a bit esoteric. This popped up when I replaced old/broken code by EqualsVerifier to check another (very simple) exception class.

# What steps will reproduce the problem?
```
EqualsVerifier.configure()
                .suppress(INHERITED_DIRECTLY_FROM_OBJECT)
                .forClass(NullPointerException.class)
                .usingGetClass()
                .verify();
```

Note: `NullPointerException` does not override `equals`, as such the default `==` implementation in `Object` is used.

# What is the code that triggers this problem?
`MutableStateFieldCheck.execute` does not check if the two objects were equal before the mutation. The boolean `equalsChanged` is wrong in the sense that, in this case, the two instances were not equal before the change, and still are not equal after the change.

# What error message or stack trace does EqualsVerifier give?

> java.lang.AssertionError: EqualsVerifier found a problem in class NullPointerException.
> -> Mutability: equals depends on mutable field detailMessage.

# What did you expect?
Test passes

# Which version of EqualsVerifier are you using?
3.1.12

# Please provide any additional information below.
Thank you for the awesome project!

**Comment by jqno** [comment]
Thanks for reporting this, and thanks for the nice words!

I can reproduce the issue as you report it. I agree that this is a bug in EqualsVerifier, and at the moment I can't offer a good workaround.

You have a good point that assuming that the two objects are equal before the mutation may not be correct; however, if I make changes based on your suggestions, a whole bunch of other test cases break. I'll have to investigate further. I'll let you know when I have a solution. I can't make any promises about when that will be. I do accept PR's however :)

**Comment by C-Otto** [comment]
Thanks. I don't need a quick fix. I can make use of `ClassAccessor.isEqualsInheritedFromObject` to skip the equals verification. Besides, this part of the project is on hold for now due to bigger issues with the code.

**Comment by jqno** [comment]
I've just released version 3.4, which fixes this. You'll still need to suppress some `Warning`s, though. Check out [this unit test](https://github.com/jqno/equalsverifier/blob/3c7593e72adb04720dba6c580bf2c30aa0179557/src/test/java/nl/jqno/equalsverifier/integration/extended_contract/DontOverrideEqualsTest.java#L40).
